src/node_ext2fs.cc

```diff
@@ -824,16 +824,13 @@ class MkDirWorker : public AsyncWorker {
 };
 X_NAN_METHOD(mkdir, MkDirWorker, 4);
 
-v8::Local<v8::Value> castUint32(long unsigned int x) {
-	return Nan::New<v8::Uint32>(static_cast<uint32_t>(x));
-}
-
-v8::Local<v8::Value> castInt32(long int x) {
-	return Nan::New<v8::Int32>(static_cast<int32_t>(x));
+v8::Local<v8::Value> getUInt64Number(long unsigned int hi, long unsigned int lo) {
+		long unsigned int size = lo | (hi << 32);
+		return Nan::New<v8::Number>(static_cast<double>(size));
```

Comment:
new `getUInt64Number()` method, which gets the full size from the high & low `inode.i_size_high` and `inode.i_size` parts respectively; and because JavaScript, that uint64 needs to be casted to a double, for Nan::New<Nan::Number> to not truncate it to 32bits

Reply:
How odd - but I guess it's fine as long as casting to a double doesn't cause any loss in precision from the uint64 value?

Reply:
Hmm, just did a bi of searching and found https://github.com/candu/node-int64-native which suggests this won't be a problem unless we have files over 8000 TB - which I guess is _fairly_ unlikely :rofl:

Reply:
Yeah, casting to `double` will lose precision for anything beyond `Number.MAX_SAFE_INTEGER`, which is `2^53 - 1` – and while still not the entire 64 bits, it's *a lot*, and should be more than enough for your everyday use

Reply:
Just because I was curious, I did a bit more searching, and was surprised to find out that Javascript doesn't support 64-bit ints (in 2018? Really?!) but https://v8project.blogspot.com/2018/05/bigint.html seems like a good step towards adding them.

Reply:
There's been a Int64 proposal around the TC somewhere iirc, but yeah, due to how Numbers work in JavaScript, it's a bit difficult to shoehorn 64bit numbers in there. If you're surprised by that, look at most other dynamic languages – most of them have IEEE 754 double floats as their number type, and are plagued by the very same issues.